src/context/FoxEthProvider/FoxEthProvider.tsx

```diff
@@ -80,156 +76,109 @@ export const FoxEthProvider = ({ children }: FoxEthProviderProps) => {
+  }, [refetchFoxEthLpAccountData])
 
-  const lpAccountFilter = useMemo(() => ({ accountId: lpAccountId ?? '' }), [lpAccountId])
-  // Use the account number of the consumer if we have it, else use account 0
-  const lpBip44Params =
-    useAppSelector(state => selectBIP44ParamsByAccountId(state, lpAccountFilter)) ??
-    adapter.getBIP44Params({ accountNumber: 0 })
+  const lpAccountFilter = useMemo(() => ({ accountId: lpAccountId }), [lpAccountId])
+  const lpBip44Params = useAppSelector(state =>
+    selectBIP44ParamsByAccountId(state, lpAccountFilter),
+  )
 
   useEffect(() => {
     // Get initial account 0 address from wallet, TODO: nuke it?
     if (wallet && adapter && lpBip44Params) {
       ;(async () => {
         if (!supportsETH(wallet)) return
         const address = await adapter.getAddress({ wallet, bip44Params: lpBip44Params })
-        // eth.getAddress and similar return a checksummed address, but state opportunities aren't
-        setLpAccountAddress(address.toLowerCase())
+        // eth.getAddress and similar return a checksummed address, but the account part of state opportunities' AccountId isn't checksummed
+        // using the checksum version would make us unable to do Txid lookup
+        setLpAccountId(toAccountId({ chainId: ethChainId, account: address }))
       })()
     }
   }, [adapter, wallet, lpBip44Params])
 
   useEffect(() => {
-    if (!lpAccountId) return
-    const lpAccountAddress = fromAccountId(lpAccountId).account
-    setLpAccountAddress(lpAccountAddress)
-  }, [lpAccountId])
-
-  useEffect(() => {
-    if (!farmingAccountId) return
-    const farmingAccountAddress = fromAccountId(farmingAccountId).account
-    setFarmingAccountAddress(farmingAccountAddress)
-  }, [farmingAccountId])
-
-  // this hook handles the data we need from the farming opportunities
-  useEffect(() => {
-    if (!readyToFetchFarmingData) return
     // getting fox-eth lp token data
-    const { getFoxFarmingContractMetrics, getFoxFarmingContractAccountData } = foxEthApi.endpoints
-    const ethAccountAddresses = ethAccountIds.map(accountId => fromAccountId(accountId).account)
-    // getting fox-eth lp token balances
-    farmingOpportunities.forEach(opportunity => {
-      const { contractAddress } = opportunity
+    ;(async () => {
+      // getting fox-eth lp token balances
+      await fetchAllStakingOpportunitiesMetadata()
       // getting fox farm contract data
-      ethAccountAddresses.forEach(accountAddress => {
-        dispatch(
-          getFoxFarmingContractMetrics.initiate({
-            contractAddress,
-            accountAddress,
-          }),
-        )
-        dispatch(
-          getFoxFarmingContractAccountData.initiate({
-            contractAddress,
-            accountAddress,
-          }),
-        )
-      })
-    })
-  }, [ethAccountIds, dispatch, readyToFetchFarmingData])
+      ;[...stakingAccountIds, ...lpAccountIds].forEach(accountId =>
```

Comment:
can do this point free if you want

Reply:
Unfortunately we have to have one more stack trace, if we use the shorthand tsc will fail since it will inherit the `index, array` positional arguments